src/XMakeCommandLine/UnitTests/CommandLineSwitches_Tests.cs

```diff
@@ -1153,6 +1156,129 @@ public void ExtractAnyLoggerParameterPickLast()
         }
 
         /// <summary>
+        /// Verifies that when the /warnaserror switch is not specified, the set of warnings is null.
+        /// </summary>
+        [Fact]
+        public void ProcessWarnAsErrorSwitchNotSpecified()
+        {
+            CommandLineSwitches commandLineSwitches = new CommandLineSwitches();
+
+            MSBuildApp.GatherCommandLineSwitches(new ArrayList(new[] { "" }), commandLineSwitches);
+
+            Assert.Null(MSBuildApp.ProcessWarnAsErrorSwitch(commandLineSwitches));
+        }
+
+        /// <summary>
+        /// Verifies that the /warnaserror switch is parsed properly when codes are specified.
+        /// </summary>
+        [Fact]
+        public void ProcessWarnAsErrorSwitchWithCodes()
+        {
+            ISet<string> expectedWarningsAsErors = new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "a", "B", "c", "D", "e" };
+
+            CommandLineSwitches commandLineSwitches = new CommandLineSwitches();
+
+            MSBuildApp.GatherCommandLineSwitches(new ArrayList(new[]
+            {
+                "\"/warnaserror: a,B , c \"", // Leading, trailing, leading and trailing whitespace
+                "/warnaserror:A,b,C",         // Repeats of different case
+                "\"/warnaserror:,    ,,\"",   // Empty items
+                "/err:D,d,E,e",               // A different source with new items and uses the short form
+                "/warnaserror:a",             // A different source with a single duplicate
+                "/warnaserror:a,b",           // A different source with  multiple duplicates
+                "/warnaserror",               // Nothing specified
+            }), commandLineSwitches);
+
+            ISet<string> actualWarningsAsErrors = MSBuildApp.ProcessWarnAsErrorSwitch(commandLineSwitches);
+
+            Assert.NotNull(actualWarningsAsErrors);
+
+            Assert.Equal(expectedWarningsAsErors, actualWarningsAsErrors, StringComparer.OrdinalIgnoreCase);
```

Comment:
I was imaging a scenario where the list of warning codes was specified in one or more response files, an environment variable and the command-line.  I figured I would just conglomerate them into one set.  If this is the case, then specifying the switch by itself doesn't add any more codes and some codes already exist so the functionality is to only treat the specified codes as errors.

This is sort of how `/p:foo=bar` works minus the ability to clear them all by specifying just `/p`.  Let's discuss this tomorrow.